### ISSUE TITLE
Request the training dataset for MarkerPose

### ISSUE THREAD

**Issue by Gjj152** [jhacsonmeza/MarkerPose#6]
Hi @jhacsonmeza 
Sorry to bother you, I'm trying to replicate your code, can you expose train_ellipsegnet.py and train_superpoint.py data sets?

**Comment by jhacsonmeza** [comment]
Hi @Gjj152
Unfortunately the dataset for `train_superpoint.py` isn't publicly available. On the other hand, you can find here https://github.com/jhacsonmeza/MarkerPose/issues/1#issuecomment-872415774 the dataset for `train_ellipsegnet.py`.
Let me know if you have any other question.

**Comment by Gjj152** [comment]
> Hi @Gjj152 Unfortunately the dataset for `train_superpoint.py` isn't publicly available. On the other hand, you can find here [#1 (comment)](https://github.com/jhacsonmeza/MarkerPose/issues/1#issuecomment-872415774) the dataset for `train_ellipsegnet.py`. Let me know if you have any other question.
Thank you for your answer. May I know the dataset structure of train_superpoint.py. Greatly appreciate!

**Comment by Gjj152** [comment]
> Hi @Gjj152 Unfortunately the dataset for `train_superpoint.py` isn't publicly available. On the other hand, you can find here [#1 (comment)](https://github.com/jhacsonmeza/MarkerPose/issues/1#issuecomment-872415774) the dataset for `train_ellipsegnet.py`. Let me know if you have any other question.

Thank you for your answer. May I know the dataset structure of train_superpoint.py. Greatly appreciate!

**Comment by Gjj152** [comment]
> Hi @Gjj152 Unfortunately the dataset for `train_superpoint.py` isn't publicly available. On the other hand, you can find here [#1 (comment)](https://github.com/jhacsonmeza/MarkerPose/issues/1#issuecomment-872415774) the dataset for `train_ellipsegnet.py`. Let me know if you have any other question.

Thank you for your answer. May I know the dataset structure of train_superpoint.py. I can make my own dataset for train_superpoint.py. Greatly appreciate!

**Comment by Gjj152** [comment]
> Hi @Gjj152 Unfortunately the dataset for isn't publicly available. On the other hand, you can find here [#1 (comment)](https://github.com/jhacsonmeza/MarkerPose/issues/1#issuecomment-872415774) the dataset for . Let me know if you have any other question.`train_superpoint.py``train_ellipsegnet.py`

Grateful appreciate!

**Comment by Gjj152** [comment]
Hi @jhacsonmeza 
Sorry to bother you again. I try to use my own dataset to run train_superpoint.py, but it always went to wrong(like the picture post down).
![image](https://github.com/jhacsonmeza/MarkerPose/assets/143152890/22049603-d4f1-484e-9e0b-15594130d863)
I made this dataset by using Azure Kinect DK, the image size is (3840,2160) . I changed your ground_truth.txt and by using this code i got the data_centers.csv(like the picture post down) 
![image](https://github.com/jhacsonmeza/MarkerPose/assets/143152890/59bbdced-2891-4a72-afd4-1d332b20e0a9)
then i changed the images and the data_centers.csv to (320,240) like this 
![image](https://github.com/jhacsonmeza/MarkerPose/assets/143152890/d412826b-51bc-460b-b7b7-f8616cbf9b7b)
By using this dataset, train_superpoint.py may ran dozens of epoch
![image](https://github.com/jhacsonmeza/MarkerPose/assets/143152890/f0393459-16e5-4b38-8d8e-e8a3b4bf75bc)
What do you think about this question.
In addition, in my dataset i used c1 to cx , am I right?
![image](https://github.com/jhacsonmeza/MarkerPose/assets/143152890/22565a37-8573-4087-80d5-bd0513be02ad)

**Comment by jhacsonmeza** [comment]
Hi @Gjj152.
The error you're getting is in this line: https://github.com/jhacsonmeza/MarkerPose/blob/91b4761e7332e56381f0236dd51bd6adfeede329/Python/modules/dataset.py#L36 and it's due to an out of bound indexation in `mask[target[:,1],target[:,0]]`. If you see the error message, the code is indexing with 240 which is a coordinate that is out of the image (considering that your image size is 320x240). What is not clear to me is if you're getting this error after many epochs or at the beginning of training. Also it's not clear in which line of `train_superpoint.py` you're getting the error.

For your second question, yes you're right: (cXx, cXy) are the coordinates of c1, and (cYx, cYy) are the coordinates of c2.

**Comment by Gjj152** [comment]
Hi @jhacsonmeza ，
After a few debugs, I found that the problem appeared in `Python/modules/transformations.py` 
https://github.com/jhacsonmeza/MarkerPose/blob/91b4761e7332e56381f0236dd51bd6adfeede329/Python/modules/transformations.py#L88
I found that some points would meet the above criteria, but would exceed the size of (320，240). For example, a bbox's left top point is (21,21) and right bottom point is (184,241) , according to your code union_wh[0]*union_wh[1] will smaller than im.size, but  it goes beyond the boundary of the image. I try to address this issue by changing 88 line. The following code is what I use to replace line 88, ' if union_wh[0]*union_wh[1] > (sz[0]-2*k)*(sz[1]-2*k):'. This solution is valid, but it will limit the target points to the range of (20,20) to (300,220), I don't know how this will affect the training of the network, **may I ask your thoughts on writing line 88?**
Finally, I wonder if you could share the specific size of this pattern's inside diameter and outside diameter.
![image](https://github.com/jhacsonmeza/MarkerPose/assets/143152890/560d2b34-fd61-4ffa-8a5e-f6c052ed7209)

**Comment by jhacsonmeza** [comment]
Hi @Gjj152, sorry for my late reply!
You're correct with the change. If you see the C++ implementation: https://github.com/jhacsonmeza/MarkerPose/blob/91b4761e7332e56381f0236dd51bd6adfeede329/C%2B%2B/src/transformations/points_trans.cpp#L68 we check if the `union_bbox` area is larger than the `restriction_bbox` area. This `restriction_bbox` is defined as:
```cpp
int k = 20;
cv::Rect restriction_bbox(k, k, sz.width-2*k, sz.height-2*k);
```
where `sz` is the sized of the resized image (320x240 in your case). When we compute `restriction_bbox.area()` we basically get `(sz.width-2*k)*(sz.height-2*k)` which is the product between the width and height of the `restriction_bbox`.
This restriction bbox is used to avoid getting points very close to the image borders, and this is why you see the points are restricted to the range (20,20) to (300,220). The change you propose is correct and is exactly what the C++ implementation does. Thank you for spotting that bug! I already solved the problem here https://github.com/jhacsonmeza/MarkerPose/commit/4794f402e784c472770bebf491ff8615fc24fb44.
It's interesting that we wasn't able to get the same error than you, maybe it's because the randomness of the transformations. Did you get it after many runs?

> Finally, I wonder if you could share the specific size of this pattern's inside diameter and outside diameter.

Inner diameter is 8mm and outer diameter is 20mm.
I'm attaching the SVG file of the target we used. You can measure all the dimensions you need there:
![marker](https://github.com/user-attachments/assets/946021b1-ab12-4876-acd1-c44ada0a2dfa)

**Comment by Gjj152** [comment]
Hi @jhacsonmeza Thank you very much for your answer. 
After completing the data set, I tried the program a dozen times, but every run failed, sometimes it reached hundreds of epcho, sometimes it stopped after dozens of epcho, maybe as you said, it result in the randomness of the transformations.

**Comment by jhacsonmeza** [comment]
I see. That's interesting. Even though the bug is clear, we didn't get any error in the multiple times we used the code. Perhaps we had luck because the transformations are applied randomly and we only trained for 300-500 epochs, we never tried training with thousands of epochs as you did. Of course the more epochs you use, the more likely it's to get the error.
However, thank you again for catching the bug! Hope you can now train without problems. And please let me know if you have any other question.

**Comment by Gjj152** [comment]
Hi, @jhacsonmeza I'm sorry to bother you again. When I finished the dataset and wanted to run robot_test.py in my own scenario, I had the problem shown in the video below. 

https://github.com/user-attachments/assets/00389c17-8847-48df-86e0-afeb93058f27

The following video demo is made use of your py superpoint.pt

https://github.com/user-attachments/assets/149ba116-ea28-4b43-92a9-6f0fa4970f34

As you can see, the coordinates of both demonstrations have deviated patterns, so I don't think it is a problem with the.pt file. According to robot_test.py, I think there may be something wrong with K1 and dist1 in my camera. I used the camera calibration app in matlab to obtain my camera parameters, so it is hard for me to imagine that there was a problem with this step.Maybe something else is wrong, please let me know what you think. In addition, I would also like to know the method you used to calibrate the camera.

**Comment by jhacsonmeza** [comment]
Hi @Gjj152, sorry for my late response.
If I'm not wrong, there's a slight difference between the OpenCV and Matlab stereo calibration. If you calibrated your system with Matlab, you need to transpose the intrinsic matrices of camera 1 and camera 2 (K1 and K2) and also the rotation R and translation t between camera 1 and camera 2. Please try this a let me know if this solves your problem.

Another alternative to check if your problem is with the calibration or with your model, could be to visualize the center of the circles predicted by your model. For example, you can print [here](https://github.com/jhacsonmeza/MarkerPose/blob/main/Python/modules/models.py#L182-L183) `c1` and `c2` and check if those coordinates are accurate, or you can even add a block of code below to visualize the points on the image:
```python
# Check first view
cv2.circle(x1, (int(c1[0,0]),int(c1[0,1])), 5, (0,0,255), -1)
cv2.circle(x1, (int(c1[1,0]),int(c1[1,1])), 5, (0,255,0), -1)
cv2.circle(x1, (int(c1[2,0]),int(c1[2,1])), 5, (255,0,0), -1)
cv2.imshow('im1',x1)
cv2.waitKey(0)
cv2.destroyAllWindows()

# Check second view
cv2.circle(x2, (int(c2[0,0]),int(c2[0,1])), 5, (0,0,255), -1)
cv2.circle(x2, (int(c2[1,0]),int(c2[1,1])), 5, (0,255,0), -1)
cv2.circle(x2, (int(c2[2,0]),int(c2[2,1])), 5, (255,0,0), -1)
cv2.imshow('im2',x2)
cv2.waitKey(0)
cv2.destroyAllWindows()
```
See that I used `x1` and `x2`which are the full resolution images.
Please let me know if you solve your problem.
